src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -174,60 +178,62 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
   // Cancel button in "Add an Amalgamating Business" is pressed.
   addAmalgamatingBusinessCancel (): void {
     this.isAddingAmalgamatingBusiness = false
+    this.setAmalgamatingBusinessesValid(true)
   }
 
   // "Add an Amalgamating Business" button is pressed.
   onAddBusinessClick (): void {
     this.isAddingAmalgamatingBusiness = true
     this.isAddingAmalgamatingForeignBusiness = false
+    this.setAmalgamatingBusinessesValid(false)
   }
 
   // "Add an Amalgamating Foreign Business" button is pressed.
   onAddForeignBusinessClick (): void {
     this.isAddingAmalgamatingBusiness = false
     this.isAddingAmalgamatingForeignBusiness = true
+    this.setAmalgamatingBusinessesValid(false)
   }
 
   async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
-    // Get the amalgamating business information
-    // Will have a different format depending on the business
-    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
-      .then((response) => {
-        return response?.data?.business
-      }).catch(() => {
-        return businessLookup
-      })
-
-    // Get the address of the amalgamating business
-    if (businessLookup.identifier && business.foundingDate) {
-      const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
-        .then((data) => {
-          // SP and GP have businessOffice instead of registeredOffice
-          return data?.registeredOffice || data?.businessOffice
-        }).catch(() => {
-          return undefined
-        })
-      if (addresses) {
-        business.officeAddress = addresses
-      }
+    let business = null
+
+    // Get the amalgamating business information, mailing address, and email if in LEAR.
+    // Otherwise, return the businesslookup object.
+    const data = await Promise.all([
+      LegalServices.fetchBusinessInfo(businessLookup.identifier),
+      AuthServices.fetchAuthInfo(businessLookup.identifier),
+      LegalServices.fetchAddresses(businessLookup.identifier)
+    ]).catch((error) => {
+      return error
+    })
+
+    if (data.length === 3) {
+      business = data[0].data?.business
+      business.businessContact = data[1].contacts[0]
+      business.officeAddress = data[2]
     }
 
+    // If the business is not null (LEAR Entity)
     // If the amalgamating businesses array is not empty, check if identifier already exists.
     // If identifier already exists, don't add the business to the array.
-    if (this.amalgamatingBusinesses.length > 0) {
-      const businessExists = this.amalgamatingBusinesses.find(function (id) {
-        return id.identifier === business.identifier
-      })
-      if (!businessExists) this.amalgamatingBusinesses.push(business)
-    } else {
-      this.amalgamatingBusinesses.push(business)
+    if (business) {
```

Comment:
Only adding if the entity is not a COLIN entity (as per my discussion with Yui).